wifiphisher/pywifiphisher.py

```diff
@@ -583,6 +601,10 @@ def start(self):
                 print ("[{0}+{1}] Selecting {0}{2}{1} interface for the deauthentication "
                        "attack\n[{0}+{1}] Selecting {0}{3}{1} interface for creating the "
                        "rogue Access Point").format(G, W, mon_iface.get_name(), ap_iface.get_name())
+                # randomize the mac addresses
+                if not args.no_mac_randomization:
+                    ap_iface.randomize_interface_mac(args.mac_ap_interface)
+                    mon_iface.randomize_interface_mac(args.mac_deauth_interface)
```

Comment:
Is there a reason you are performing the argument check inside the method?

I would expect something like:

```python
if not args.no_mac_randomization:
   if not args.mac_ap_interface:
      ap_iface.randomize_interface_mac()